src/lib/y2storage/autoinst_profile/drive_section.rb

```diff
@@ -309,6 +320,26 @@ def init_from_md(md)
         true
       end
 
+      # Method used by {.new_from_storage} to populate the attributes when
+      # cloning a Bcache.
+      #
+      # @param bcache [Y2Storage::Bcache] Bcache device
+      # @return [Boolean]
+      def init_from_bcache(bcache)
+        @type = :CT_BCACHE
+        @device = bcache.name
+        collection =
+          if bcache.filesystem
+            [bcache]
+          else
+            bcache.partitions
+          end
+        @partitions = partitions_from_collection(collection)
+        @enable_snapshots = enabled_snapshots?(collection.map(&:filesystem).compact)
```

Comment:
Not exactly related to this code review, but this makes me wonder whether it would make sense to use snapshots for a root filesystem on top of a bcache device. I mean, such setup will likely need a separate "/boot", since I don't believe Grub can boot from the bcache device. And with a separate "/boot", grub will likely not be able to run from a previous snapshot.